### ISSUE TITLE
feat: add support for multi-module repos

### ISSUE THREAD

**Issue by dolmen** [actions/setup-go#371]
**Description:**
Add support for caching dependencies for repositories made of multiple Go modules.

Example:
* Repo [dolmen-go/mylogin-driver](https://github.com/dolmen-go/mylogin-driver/) has Go modules [`github.com/dolmen-go/mylogin-driver`](https://github.com/dolmen-go/mylogin-driver/blob/master/go.mod) and [`github.com/dolmen-go/mylogin-driver/cmd/mylogin-query`](https://github.com/dolmen-go/mylogin-driver/blob/master/cmd/mylogin-query/go.mod)

**Justification:**
Each Go module has its own set of dependencies. Dependencies caching should cover the union of the dependencies of all Go modules of the repo.

From the [documentation](https://github.com/actions/setup-go/blob/dd84a9531a6f8e72c321f2aa3b9048ed359670e4/README.md#caching-dependency-files-and-build-outputs), it looks like a single Go module is supported.

**Are you willing to submit a PR?**
No. I'm not familiar with GitHub Actions development.

**Comment by dsame** [comment]
Hello @dolmen, thanks for the suggestion, we are starting to investigate it

**Comment by dolmen** [comment]
Design idea: allow to path an array of path for the `cache-dependency-path` option as an alternative to a string.

**Comment by dsame** [comment]
This feature resembles https://github.com/actions/setup-node/pull/735

**Comment by dsame** [comment]
Hello @dolmen ,

`cache-dependency-path` accepts both list and glob syntax:

```
          cache-dependency-path: |
             hello/go.sum 
             greeting/go.sum
```

```
          cache-dependency-path: "**/*.sum"
```

Sample workflow: https://github.com/akv-demo/test-setup-go/blob/cache-multi/.github/workflows/blank.yml
Sample build: https://github.com/akv-demo/test-setup-go/actions/runs/5799499947/job/15719631207

Did this answer help?

**Comment by dolmen** [comment]
Not yet tested, but it seems to fit my need.

However, what about making `**/go.sum` the default?

**Comment by dsame** [comment]
@dolmen i do not think it is good idea to allow implicit dependencies. I believe the user of the action should have clear understanding of where the dependencies come from.

**Comment by dsame** [comment]
@dolmen i am going to close the issue because the problem is resolved without the implementing requested feature, but please feel free to reopen it or create new one in case if the problem still exists

**Comment by artemgavrilov** [comment]
Hi @dsame , how about make documentation regarding this a little bit more clear?
https://github.com/artemgavrilov/setup-go/pull/1
